docs/english/tutorials/random-fact-generator/random-fact-generator.md

```diff
@@ -0,0 +1,297 @@
+      "description": "Runs useless fact step",
+      "input_parameters": {},
+      "output_parameters": {
+        "message": {
+          "type": "string",
+          "title": "Fact",
+          "description": "A random useless fact",
+          "is_required": true,
+          "name": "message"
+        }
+      }
+    }
+  }
+}
+```
+
+Note the app manifest includes a function named `useless_fact_step` and declares the function returns one value named “Fact” as its output. We'll get to this later.
+
+### Save tokens
+
+Once your app has been created, scroll down to **App-Level Tokens** on the **Basic Information** page and create a token that requests the [`connections:write`](/reference/scopes/connections.write) scope. This token will allow you to use [Socket Mode](/apis/events-api/using-socket-mode), which is a secure way to develop on Slack through the use of WebSockets. Save the value of your app token and store it in a safe place. Also, save the **Signing Secret** from the **App Credentials** section of this page. We’ll use these later.
+
+### Install app in workspace
+
+Still in the app settings, navigate to the **Install App** page in the left sidebar. Install your app. When you press **Allow**, this means you’re agreeing to install your app with the permissions that it’s requesting. Copy the bot token that you receive and store it in a safe place for subsequent steps.
+
+Now that you’ve completed the app setup, it’s time to write some code!
+
+### Install Node and set up the project
+
+Open your terminal or command prompt. First, check to see that you have Node.js installed and it is a recent long-term support (LTS) version by typing the following command.
+
+```sh
+node -v
+```
+
+If you get an error or if it’s an older version than what’s available for download on the [Node.js website](https://nodejs.org/), take a moment to install the latest version.
+
+Next, create a new directory for the project named `useless-fact-app` and change to the new directory.
+
+```sh
+mkdir useless-fact-app
+cd useless-fact-app
+```
+
+For Windows, the command looks like this:
+
+```powershell
+md useless-fact-app
+```
+
+Initialize a new Node.js project using the following command; answer each question with defaults.
+
+```sh
+npm init
+```
+
+Our project is going to use two packages: the Slack [Bolt for JavaScript](/tools/bolt-js/) framework and [Axios](https://www.npmjs.com/package/axios), a popular HTTP client for making API calls. Install these dependencies with the following command.
+
+```sh
+npm install @slack/bolt axios
+```
+
+Open the project in VS Code or your favorite code editor, then create a new file in the project folder named `.env`. Open the file and paste the following, replacing the placeholders with the values you saved earlier. 
+
+```sh
+SLACK_SIGNING_SECRET={your-app-signing-secret}
+SLACK_BOT_TOKEN={your-bot-token-xoxb-1234}
+SLACK_APP_TOKEN={your-app-token-xapp-1234}
+PORT=3000
+```
```

Comment:
````suggestion
```txt
SLACK_BOT_TOKEN={your-bot-token-xoxb-1234}
SLACK_APP_TOKEN={your-app-token-xapp-1234}
```
````

💸 suggestion: Related to signing secrets and socket mode!